test/llilc_run.py

```diff
@@ -0,0 +1,105 @@
+﻿#!/usr/bin/env python
+#
+# title           : llilcrun.py
+# description     : Run a managed application using LLILC jit.
+#   
+
+import argparse
+import os
+import subprocess
+import sys
+
+llilcverbose = False
+
+def QuoteArg(arg):
+    '''Strip off any enclosing single quotes and enclose in double quotes'''
+    arg = '"' + arg.strip("'").strip('"') + '"'
+    return arg
+
+def UnquoteArg(arg):
+    ''' Remove single and double quotes from front and back of arg'''
+    return arg.strip("'").strip('"')
+
+def RunCommand(command):
+    ''' Run a command and return its exit code, optionally echoing it.'''
+    global llilcverbose
+    if llilcverbose:
+        sys.stderr.write (command + '\n')
+        sys.stdout.write (command + '\n')
+    error_level = subprocess.call(command, shell=True)
+    return error_level
+
+def main(argv):
+    '''
+    main method of script. arguments are script path and remaining arguments.
+    '''
+    global llilcverbose
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--llilc-dump-level', type=str, choices={'summary', 'verbose'}, 
```

Comment:
I didn’t see this comment until after I merged, but as I said in a separate message, I think using ‘--' to separate the script args from the client args is the best option as it does not involve adding an extra parsing step. In any case, we can discuss that in the follow-on version.

From: Michelle McDaniel [mailto:notifications@github.com]
Sent: Friday, August 07, 2015 12:53
To: dotnet/llilc <llilc@noreply.github.com>
Cc: Rich Ford <richford@microsoft.com>
Subject: Re: [llilc] Add support for building Roslyn (and other apps) with LLILC. (#742)


In test/llilc_run.py<https://na01.safelinks.protection.outlook.com/?url=https%3a%2f%2fgithub.com%2fdotnet%2fllilc%2fpull%2f742%23discussion_r36554770&data=01%7c01%7crichford%40microsoft.com%7c51061f3c4de346332c5908d29f61ccc1%7c72f988bf86f141af91ab2d7cd011db47%7c1&sdata=nVsvrw3o6zfQ6INVu6M4adnNN8DxRwODe6PDVMPT0fY%3d>:

> +def RunCommand(command):

> +    ''' Run a command and return its exit code, optionally echoing it.'''

> +    global llilcverbose

> +    if llilcverbose:

> +        sys.stderr.write (command + '\n')

> +        sys.stdout.write (command + '\n')

> +    error_level = subprocess.call(command, shell=True)

> +    return error_level

> +

> +def main(argv):

> +    '''

> +    main method of script. arguments are script path and remaining arguments.

> +    '''

> +    global llilcverbose

> +    parser = argparse.ArgumentParser()

> +    parser.add_argument('--llilc-dump-level', type=str, choices={'summary', 'verbose'},

I'm not sure I like that scheme. In fact, looking at the script, I didn't realize that was happening, suggesting that it is not really intuitive. I would be more comfortable if any arguments being passed on to the program being run were explicitly passed. Like having a --program-args or --app-args option that took a semicolon separated list of options and then parsing those args and sending them to the program being called.

—
Reply to this email directly or view it on GitHub<https://na01.safelinks.protection.outlook.com/?url=https%3a%2f%2fgithub.com%2fdotnet%2fllilc%2fpull%2f742%2ffiles%23r36554770&data=01%7c01%7crichford%40microsoft.com%7c51061f3c4de346332c5908d29f61ccc1%7c72f988bf86f141af91ab2d7cd011db47%7c1&sdata=NU70KRaNfVwzSG%2bZLkDwk%2favgxNncMsuFiPDU39ncSw%3d>.